records_mover/db/mysql/mysql_db_driver.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional, Tuple, Union
+
+
+logger = logging.getLogger(__name__)
+
+
+class MySQLDBDriver(DBDriver):
+    # def __init__(self,
+    #              db: Union[sqlalchemy.engine.Engine, sqlalchemy.engine.Connection],
+    #              **kwargs) -> None:
+    #     super().__init__(db)
+
+    # https://www.postgresql.org/docs/10/datatype-numeric.html
+    def integer_limits(self,
+                       type_: sqlalchemy.types.Integer) ->\
+            Optional[Tuple[int, int]]:
+        if isinstance(type_, sqlalchemy.dialects.mysql.TINYINT):
+            return (INT8_MIN, INT8_MAX)
+        elif isinstance(type_, sqlalchemy.sql.sqltypes.SMALLINT):
+            return (INT16_MIN, INT16_MAX)
+        elif isinstance(type_, sqlalchemy.dialects.mysql.MEDIUMINT):
+            return (INT24_MIN, INT24_MAX)
+        elif isinstance(type_, sqlalchemy.sql.sqltypes.INTEGER):
+            return (INT32_MIN, INT32_MAX)
+        elif isinstance(type_, sqlalchemy.sql.sqltypes.BIGINT):
+            return (INT64_MIN, INT64_MAX)
+        return super().integer_limits(type_)
+
+    def fp_constraints(self,
+                       type_: sqlalchemy.types.Float) ->\
+            Optional[Tuple[int, int]]:
+        if isinstance(type_, sqlalchemy.dialects.mysql.DOUBLE):
+            return (64, FLOAT64_SIGNIFICAND_BITS)
+        elif isinstance(type_, sqlalchemy.sql.sqltypes.FLOAT):
+            return (32, FLOAT32_SIGNIFICAND_BITS)
+        return super().fp_constraints(type_)
+
+    def type_for_integer(self,
+                         min_value: Optional[int],
+                         max_value: Optional[int]) -> sqlalchemy.types.TypeEngine:
+        """Find correct integral column type to fit the given min and max integer values"""
+
+        if min_value is not None and max_value is not None:
+            pass
+            if min_value >= INT8_MIN and max_value <= INT8_MAX:
+                return sqlalchemy.dialects.mysql.TINYINT()
+            elif min_value >= UINT8_MIN and max_value <= UINT8_MAX:
+                return sqlalchemy.dialects.mysql.TINYINT(unsigned=True)
+            elif min_value >= INT16_MIN and max_value <= INT16_MAX:
+                return sqlalchemy.sql.sqltypes.SMALLINT()
+            elif min_value >= UINT16_MIN and max_value <= UINT16_MAX:
+                return sqlalchemy.dialects.mysql.SMALLINT(unsigned=True)
+            elif min_value >= INT32_MIN and max_value <= INT32_MAX:
+                return sqlalchemy.sql.sqltypes.INTEGER()
+            elif min_value >= UINT32_MIN and max_value <= UINT32_MAX:
+                return sqlalchemy.dialects.mysql.INTEGER(unsigned=True)
+            elif min_value >= INT64_MIN and max_value <= INT64_MAX:
+                return sqlalchemy.sql.sqltypes.BIGINT()
+            elif min_value >= UINT64_MIN and max_value <= UINT64_MAX:
+                return sqlalchemy.dialects.mysql.BIGINT(unsigned=True)
+    #         else:
+    # TODO: Figure out a test that needs the below
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker